src/DynamoCore/Updates/UpdateManager.cs

```diff
@@ -551,6 +567,43 @@ public static bool TryGetConfigFilePath(out string filePath)
         public IDynamoLookUp DynamoLookUp { get; set; }
     }
 
+    /// <summary>
+    /// Specifies the disable update requests settings.
+    /// </summary>
+    public class InstallButtonSettings : IinstallButtonSettings
```

Comment:
Hi @ZiyunShang thanks for this PR!

I feel it might be useful to make the class `UpdateManagerConfiguation` implement this interface and its old interface like this

``` c#
public Interface IDisableUpdateConfig {
Boolean DisableUpdates {get;set;}
}
```


``` c#
   //TODO - combine these interfaces when dynamo 3.0 is released.
    public class UpdateManagerConfiguration : IUpdateManagerConfiguration, IDisableUpdateConfig {
...
}

```

* This way the disable setting is part of the configuration so it can be modified without rebuilding if required or for manual testing. I think it might require a bit more work just to make sure it can be loaded correctly if serialized.

* In either case I think this PR needs a simple test added to the updateManager tests here:
https://github.com/DynamoDS/Dynamo/blob/master/test/DynamoCoreTests/UpdateManagerTests.cs

Adding the interface like this should not break binary compatibility. (your way would not either, this is just an idea to add it to the config instead of adding a new property. If only adding the property I think it can be as simple as a boolean?
What do you think?

Reply:
Hi @mjkkirschner , I think it would be nice to put it directly into this interface as a property, but I don't know if it would have any other effect. I have changed the new interface to a property and put in IUpdateManagerConfiguration, and try to add a new test to cover it. But I don't know how to run the test to verify whether my new test is valid or not.